### PR TITLE
Enable O3 optimizations

### PR DESCRIPTION
**Pull request by AJenbo**
Enabling O3 get the FPS in dungions up to 23-27 on RG300

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -447,6 +447,8 @@ if(HAIKU)
   target_link_libraries(${BIN_TARGET} PRIVATE network)
 endif()
 
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3")
```
Comment (glebm) [inline CMakeLists.txt]: C_FLAGS as well?

Also for nightly:
`CMAKE_${lang}_FLAGS_RELWITHDEBINFO`

**Comment by MrHuu** [comment]
set(CMAKE_BUILD_TYPE "Release") already defines -O3 in my experience.

While looking into the matter, i was reading this:
http://www.brianlheim.com/2018/04/09/cmake-cheat-sheet.html#build-types

The 3ds build already has it set in the generated flags.make files.

**Comment by AJenbo** [comment]
There is a definite difference for RG300 so it must at least not be the default there. @MrHuu Do you have a better suggestion for how to make sure that O3 is enabled for release builds?

**Comment by AJenbo** [comment]
Indeed this is enabled on normal builds but not for retrofw cross-compiling. Also RG350 builds at -O2.

**Comment by AJenbo** [comment]
Merged via https://github.com/diasurgical/devilutionX/commit/96be58f870e8d58cd3464fbb28bc959265050d2d (only affecting retrofw)

**Pull request closed without merging**